rosidl_generator_cpp/resource/msg__struct.hpp.em

```diff
@@ -219,6 +256,43 @@ for field in spec.fields:
 using @(spec.base_type.type) =
     @(cpp_full_name)<std::allocator<void>>;
 
+@[if need_primitive_init]@
+template<class ContainerAllocator>
+void @(spec.base_type.type)_primitive_initializer(rosidl_generator_cpp::MessageInitialization _init, @(spec.base_type.type)_<ContainerAllocator> * _obj)
```

Comment:
Why `primitive` in the function name? Doesn't it also handle initialization of arrays and potentially complex types?